clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -324,32 +329,54 @@ static bool selectBfloatLibs(const llvm::Triple &Triple, const Compilation &C,
       }
     }
 
-    UseNative = false;
-
-    // Check for intel_gpu_pvc as the target
-    if (Arg *SYCLTarget = Args.getLastArg(options::OPT_fsycl_targets_EQ)) {
-      if (SYCLTarget->getValues().size() == 1) {
-        StringRef SYCLTargetStr = SYCLTarget->getValue();
-        if (SYCLTargetStr == "intel_gpu_pvc")
-          UseNative = true;
-      }
-    }
-
-    auto checkBF = [](StringRef Device) {
-      return Device.starts_with("pvc") || Device.starts_with("ats");
-    };
+    // We need to select fallback/native bfloat16 devicelib in AOT compilation
+    // targetting for Intel GPU devices. Users have 2 ways to apply AOT,
+    // 1). clang++ -fsycl -fsycl-targets=spir64_gen -Xs "-device pvc,...,"
+    // 2). clang++ -fsycl -fsycl-targets=intel_gpu_pvc,...
+    // We assume that users will only apply either 1) or 2) and won't mix the
+    // 2 ways in their compiling command.
```

Comment:
Given this assumption, can we emit a diagnostic when this s combination occurs?

Reply:
Hi, @mdtoguchi 
The assumption is listed here because compiler has already emitted error message about such mix use. For example, if we mix using like:
clang++ -fsycl -fsycl-targets=spir64_gen,intel_gpu_dg1 -Xs "-device pvc" test.cpp
Compilation will abort and emit:
clang++: error: unsupported option '-device' for target 'intel_gpu_dg1'

Thanks very much.

Reply:
That error isn't guaranteed.  For instance:
`clang++ -fsycl -fsycl-targets=spir64_gen,intel_gpu_dg1 -Xsycl-target-backend=spir64_gen "-device pvc" test.cpp`
This will compile, and create 2 different device binaries.  The use of `-Xs <arg>` is a more general usage where `<arg>` is passed to all device compilations, while you can steer specific args using `-Xsycl-target-backend=target <arg>`

Reply:
Hi, @mdtoguchi 
Thanks for pointing out this, we need to take care of the scenario you mentioned, native bfloat16 devicelib can only be used when all GPU AOT devices support native bfloat16 conversion.